Users/binah/access-control/access-control.cs

```diff
@@ -0,0 +1,320 @@
+
+    [DefaultValue("/unban")]
+    [Tooltip("Command to remove a user from banlist")]
+    [DisplayName("Chat Command")]
+    public readonly string UnBanCommand = "/unban";
+
+    [DisplayName("Banned Destination")]
+    [Tooltip("The destination to send banned users.")]
+    [DefaultValue("https://atlas.sansar.com/experiences/katylina/you-have-been-expelled")]
+    public string BannedDestination;
+
+    [DefaultValue(true)]
+    public bool DebugLogging;
+    #endregion
+
+    private RigidBodyComponent _rb;
+
+    private List<string> Admins = new List<string>(); // handle
+    private List<string> Banned = new List<string>(); 
+
+    public override void Init()
+    {
+        // get the holding container rigid body
+        if (!ObjectPrivate.TryGetFirstComponent(out _rb))
+        {
+            if (DebugLogging) Log.Write("Script not running on an object with a physics volume!");
+        } else
+        {
+            if (DebugLogging) Log.Write("RigidBody found!");
+        }
+
+        // Subscribe to Add User events
+        // ---------------------------------------------->start coroutine TrackUser send it UserData
+        ScenePrivate.User.Subscribe(User.AddUser, SessionId.Invalid, (UserData data) => StartCoroutine(TrackUser, data.User), true);
+        if (DebugLogging) Log.Write("Subscribed to TrackUser");
+
+        //Add admins, scene owner is automatically added
+        Admins.AddRange(AdminHandles.Trim().ToLower().Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries));
+        Admins.Add(ScenePrivate.SceneInfo.AvatarId);
+
+        if(Admins.Count == 1)
+        {
+            //failsafe to not lock visitors into a cube when no admin is around
+            DoorsOpen = true;
+        }
+
+        // listen for commands
+        ScenePrivate.Chat.Subscribe(0, null, ShowLogCommand);
+        ScenePrivate.Chat.Subscribe(0, null, ToggleDoors);
+        ScenePrivate.Chat.Subscribe(0, null, OnBanCommand);
+        ScenePrivate.Chat.Subscribe(0, null, OnUnBanCommand);
+    }
+
+
+    public class Visitor
+    {
+        public string Name { get; internal set; }
+        public string Handle { get; internal set; }
+        public AgentPrivate Agent { get; internal set;  }
+        public bool isAdmin { get; internal set; }
+        public bool isBanned { get; internal set; }
+    }
+
+    private Dictionary<string, Visitor> Visitors = new Dictionary<string, Visitor>();
+
+    bool IsAdmin(AgentPrivate agent)
+    {
+        if (DebugLogging) Log.Write(agent.AgentInfo.Handle + " IsAdmin: " + Admins.Contains(agent.AgentInfo.Handle.ToLower()));
+        return agent != null && Admins.Contains(agent.AgentInfo.Handle.ToLower());
+    }
+
+    bool IsBanned(AgentPrivate agent)
+    {
+        if (DebugLogging) Log.Write(agent.AgentInfo.Handle + " IsBanned: " + Banned.Contains(agent.AgentInfo.Handle.ToLower()));
+        return Banned.Contains(agent.AgentInfo.Handle.ToLower());
```

Comment:
You might want to exclude admins in this check, incase an admin accidentally bans themselves.

Reply:
Good point!